crucible-syntax/src/Lang/Crucible/Syntax/Concrete.hs

```diff
@@ -1749,49 +1842,53 @@ termStmt' retTy =
            -> m (Ctx.Assignment (Atom s) argTypes)
         go Ctx.AssignEmpty = emptyList *> pure Ctx.empty
         go (Ctx.AssignExtend tys ty) =
-          depCons (located (reading (check ty))) $
+          depCons (located (reading (check ty hooks))) $
             \(Posd loc arg) ->
                Ctx.extend <$> go (Ctx.viewAssign tys) <*> eval loc arg
 
     err :: m (TermStmt s ret)
     err =
-      do Posd loc e <- unary Error_ (located (reading (check (StringRepr UnicodeRepr))))
+      do Posd loc e <- unary Error_ (located (reading (check (StringRepr UnicodeRepr) hooks)))
          ErrorStmt <$> eval loc e
 
     out :: m (TermStmt s ret)
     out = followedBy (kw Output_) $
           do -- commit
              depCons lambdaLabel $
                \(Pair argTy lbl) ->
-                 depCons (located (reading (check argTy))) $
+                 depCons (located (reading (check argTy hooks))) $
                    \(Posd loc arg) ->
                      emptyList *>
                        (Output lbl <$> eval loc arg)
 
 
 
-data Rand s t = Rand (AST s) (E s t)
+data Rand s ext t = Rand (AST s) (E s ext t)
 
 
 
 
 --------------------------------------------------------------------------
 
 -- | Any CFG, regardless of its arguments and return type, with its helpers
-data ACFG :: * where
-  ACFG :: forall (s :: *) (init :: Ctx CrucibleType) (ret :: CrucibleType) .
+data ACFG ext :: * where
+  ACFG :: forall (s :: *) (init :: Ctx CrucibleType) (ret :: CrucibleType) ext .
+          ( LCCE.IsSyntaxExtension ext ) =>
           CtxRepr init -> TypeRepr ret ->
-          CFG () s init ret ->
-          ACFG
+          CFG ext s init ret ->
+          ACFG ext
 
-deriving instance Show ACFG
+deriving instance Show (ACFG ())
```

Comment:
Very minor - is it possible to make this cover the generic case? Probably not a huge deal if not

Reply:
Yep.  Done